### PR TITLE
Simplified release notes templates [skip ci]

### PR DESCRIPTION
**Pull request by jouni**
Major overhaul on the release notes contents. Tried to focus on the most relevant information.

### CODE REVIEW

**Comment by jouni** [comment]
@ZheSun88, how can I test these locally? I looked at the scripts/generator/README.md, but that command didn't work for me.

**Comment by jouni** [comment]
Where is the contents for `{{changesSincePrevious}}` defined when doing a release? I’d like to propose shared guidelines for those as well (for example, prefer headings+paragraphs instead of nested lists).

**Comment by ZheSun88** [comment]
@jouni   generating all release note can be done by using `node scripts/generator/generate.js --platform=<version> --versions=versions.json` under the root of platform project.  

for the {{changesSincePrevious}}, the same command can be used here too. I tested with 22.0 branch, but this somehow not working in your branch, i will check separately.

File: scripts/generator/templates/template-release-notes.md [hunk 1]
```diff
@@ -1,187 +1,132 @@
...
+- Maven plugin ({{platform}})
+- Gradle plugin ({{platform}})
+- OSGi plugin ([{{core.flow-osgi.javaVersion}}](https://github.com/vaadin/osgi/releases/tag/{{core.flow-osqi.javaVersion}}))
+- Quarkus plugin ([{{core.vaadin-quarkus.javaVersion}}](https://github.com/vaadin/quarkus/releases/tag/{{core.vaadin-quarkus.javaVersion}}))
+
+
+## <a id="_upgrading_guides"></a> Upgrading guides
+
+- [Upgrading Flow to Vaadin 23](https://vaadin.com/docs/latest/flow/upgrading/changes/#changes-in-vaadin-23)
+- [Upgrading Fusion to Vaadin 23](https://vaadin.com/docs/latest/fusion/upgrading/changes/#changes-in-vaadin-22)
```
Comment (ZheSun88) [inline scripts/generator/templates/template-release-notes.md:30]: >[Upgrading Fusion to Vaadin 23](https://vaadin.com/docs/latest/fusion/upgrading/changes/#changes-in-vaadin-22)

this link refers to changes in vaadin 22. i think it should be vaadin 23, right?

File: scripts/generator/templates/template-release-notes.md [hunk 2]
```diff
@@ -1,187 +1,132 @@
...
-Read more about upgrading to vaadin 23 from [vaadin.com](https://vaadin.com/docs/latest/guide/upgrading).
-
-
-# Supported Technologies
-## Operating Systems
-Development is supported with the following operating systems:
-- Windows
-- Linux
-- macOS
+Vaadin 23 is supported for one month after Vaadin 24 has been released ([release model](https://vaadin.com/roadmap)).
```
Comment (ZheSun88) [inline scripts/generator/templates/template-release-notes.md:36]: Vaadin 23 is a LTS, so we will need to update the statement here.

**Review by ZheSun88 — COMMENTED** [review]
should we mention the theme support separately as before? 
and maybe put a link to the starter page for users?

File: scripts/generator/templates/template-release-notes.md (hunk 1 above)
Comment (jouni) [reply scripts/generator/templates/template-release-notes.md:30]: Yes, thanks, I apparently missed that.

Comment (jouni) [reply scripts/generator/templates/template-release-notes.md:30]:
```suggestion
- [Upgrading Fusion to Vaadin 23](https://vaadin.com/docs/latest/fusion/upgrading/changes/#changes-in-vaadin-23)
```

File: scripts/generator/templates/template-release-notes.md (hunk 2 above)
Comment (jouni) [reply scripts/generator/templates/template-release-notes.md:36]: Ah, right. I did remove the mention about 14 LTS, but didn't pay attention how this sentence was formulated.

Comment (ZheSun88) [reply scripts/generator/templates/template-release-notes.md:36]: no worries.. that is not your fault.. the original content is wrong too.

**Comment by jouni** [comment]
> should we mention the theme support separately as before?

Hmm, not sure, but my instinct was "no". Those are part of the web-components monorepo, so it feels redundant in some sense.

> and maybe put a link to the starter page for users?

I feel that’s covered by the "Docs" link in the beginning. My assumption is that we can expect most users to assume they’ll find getting started instructions there. @tarekoraby, what do you think?

File: scripts/generator/templates/template-release-notes.md (hunk 2 above)
Comment (jouni) [reply scripts/generator/templates/template-release-notes.md:36]:
```suggestion
<!-- New LTS:

Vaadin 23 is an LTS release, and is supported for 5 years, with extended support options available ([release model](https://vaadin.com/roadmap)).

-->

<!-- Non-LTS:

Vaadin 24 is supported for one month after Vaadin 25 has been released ([release model](https://vaadin.com/roadmap)).

-->
```

Comment (jouni) [reply scripts/generator/templates/template-release-notes.md]: I added two options to choose from. Need to uncomment the one to use.

**Review by ZheSun88 — APPROVED** [review]

**Comment by tarekoraby** [comment]
Sorry @jouni, I missed the mention earlier. I would have preferred if we added a link to vaadin.com/start. But we can do it in a separate PR perhaps.

**Comment by jouni** [comment]
Yes, please, do add what you need. This is by no means set in stone.

**Pull request merged**